actions.schema.json

```
@@ -0,0 +1,64 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "actions.schema.json#",
+  "title": "Action Definition",
+  "description": "Describes an action with its scope and steps to perform.",
+  "allowComments": true,
+  "type": "object",
+  "additionalProperties": false,
+  "required": [
+    "scopeQuery",
+    "steps"
+  ],
+  "properties": {
+    "scopeQuery": {
+      "description": "A Sourcegraph search query to generate a list of repositories over which to run the action. Use 'src actions scope-query' to see which repositories are matched by the query.",
+      "type": "string",
+      "minLength": 1
+    },
+    "steps": {
+      "description": "A list of action steps to execute in each repository.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": ["type"],
+        "additionalProperties": false,
+        "properties": {
+          "type": {
+            "description": "Can be either \"command\", which executes the step in the native environment (OS) of the machine where 'src actions exec' is executed, or \"docker\" which runs a container with the repository contents mounted in at `/work`. Note that local images (not from a Docker registry) must be built manually prior to executing.",
```

```suggestion
            "description": "Can be either \"command\", which executes the step in the native environment (OS) of the machine where 'src actions exec' is executed, or \"docker\" which runs a container with the repository contents mounted in at `/work`. Note that images from the Docker registry must be pulled manually prior to executing.",
```

I think it's clear that local _images_ must be built before (where would you get the image name from, if not from `docker build`?) See my other comment for the `dockerfile` attribute.